### PR TITLE
8282805: Javadoc for restricted method does not reflect updates to --enable-native-access

### PR DESCRIPTION
**Pull request by mcimadamore**
This simple patch updates the javadoc comments for restricted methods to reflect the updated behavior of the `--enable-native-access` flag.
All the `@throws` tags in restricted methods are updated, as well as the toplevel package javadoc section on restricted methods.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed

### Issue
 * [JDK-8282805](https://bugs.openjdk.java.net/browse/JDK-8282805): Javadoc for restricted method does not reflect updates to --enable-native-access

### Reviewers
 * [Athijegannathan Sundararajan](https://openjdk.java.net/census#sundar) (@sundararajana - Committer) ⚠️ Review applies to 324db9c10dd3a187ae42d41d7c328e364e12dc5c

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/panama-foreign pull/665/head:pull/665` \
`$ git checkout pull/665`

Update a local copy of the PR: \
`$ git checkout pull/665` \
`$ git pull https://git.openjdk.java.net/panama-foreign pull/665/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 665`

View PR using the GUI difftool: \
`$ git pr show -t 665`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/panama-foreign/pull/665.diff">https://git.openjdk.java.net/panama-foreign/pull/665.diff</a>

</details>

### CODE REVIEW

**Comment by FrauBoes** [comment]
LGTM

File: src/java.base/share/classes/java/lang/foreign/package-info.java [hunk 1]
```diff
@@ -226,7 +226,10 @@
  * memory when attempting to dereference said segment. For these reasons, it is crucial for code that calls a restricted method
  * to never pass arguments that might cause incorrect binding of foreign data and/or functions to a Java API.
  * <p>
- * Access to restricted methods is <em>disabled</em> by default; to enable restricted methods, the command line option
- * {@code --enable-native-access} must mention the name of the caller's module.
+ * Access to restricted methods can be controlled using the command line option {@code --enable-native-access=M1,M2, ... Mn},
+ * where {@code M1}, {@code M2}, {@code ... Mn} are module names (for the unnamed module, the special value {@code ALL-UNNAMED}
+ * can be used). If this option is specified, access to restricted methods is only granted to the modules listed by that
+ * option. If no command line option is specified, access to restricted method is enabled for all modules, but
+ * access to restricted methods will result in runtime warnings.
```
Comment (sundararajana) [inline src/java.base/share/classes/java/lang/foreign/package-info.java:233]: “If this option is specified…” for the positive case
“If no command line option is specified” for the negative case

Should it be “If this option is not specified..” for negative case?

**Review by sundararajana — APPROVED** [review]
LGTM

**Pull request closed without merging**